src/server/game/Spells/Spell.cpp

```diff
@@ -2130,6 +2166,10 @@ void Spell::AddUnitTarget(Unit* target, uint32 effectMask, bool checkIfValid /*=
         // Calculate minimum incoming time
         if (!m_delayMoment || m_delayMoment > targetInfo.TimeDelay)
             m_delayMoment = targetInfo.TimeDelay;
+
+        if (unitCaster && targetInfo.MissCondition != SPELL_MISS_EVADE) {
+            target->m_Events.AddEvent(new ProcImpactDelayed(unitCaster, targetGUID, m_spellInfo, targetInfo.IsPVPEnabling()), target->m_Events.CalculateTime(Milliseconds(targetInfo.TimeDelay)));
```

Comment:
Is it safe to store unitCaster pointer in an event that will be executed later on, with a chance of unitCaster having logged out in the meantime ?

Reply:
@jackpoz i guess that was a rhetorical question? :D

Reply:
Confirmed. I haven't had the bandwidth to do any TC commits in the last couple days.

Reply:
Made the suggested changes where target is the first parameters.

I'm going to add one more specific test, just to be put my mind at ease (I think my brain might still stuck in some wrong idea of the checks and statuses): Log a character out while a projectile is in flight towards them.